homeassistant/components/prometheus/__init__.py

```diff
@@ -328,9 +330,9 @@ def _handle_climate(self, state):
             if self._climate_units == TEMP_FAHRENHEIT:
                 current_temp = fahrenheit_to_celsius(current_temp)
             metric = self._metric(
-                "current_temperature_c",
+                "climate_current_temp_celsius",
```

Comment:
I disagree with the `current` in the name, we are talking about a timeseries after all.

Reply:
I agree that the name is a bit weird, even though "current temperature at the time of measurement" does make *some* sense as a time series. Is there an alternative name you would like to suggest? I do believe we need to explicitly differentiate between measured temperature and target temperature, since most thermostats will export both.

Reply:
How about:

- climate_temperature_celsius
- climate_target_temperature_celsius

Reply:
Good point about `temp`, I agree.

I would, however, prefer to make metric semantics more explicit rather than calling one of them "temperature". There are two reasons for that:
1. I think names should be clear without context. While it might be obvious that "climate_temperature_celsius" means "current temperature" when you see another metric called "climate_target_temperature_celsius", if you only see "climate_temperature_celsius" (on a graph, or in alert definition) you don't really know whether it's target or measured temperature.
2. Currently the *target* temperature metric has a more generic name (`temperature_c`) and making the measured temperature metric more generic might be confusing.

Do you think `climate_measured_temperature_celsius` is better than `climate_current_temperature_celsius`? I still like `current` better, but `measured` is the only alternative I can come up with.

Reply:
The word you are looking for is probably `actual`.

Reply:
On the climate entity, we use "temperature" to mean the current temperature and qualify the other temperatures. It seems.reasonable to do the same here to make the correspondence clear.

Reply:
Thanks for chiming in, @mdz. Could you please clarify whether the "temperature" is used to mean current temperature in the climate entity code, or if this is a more broad user-centered convention?

For reasons mentioned above (to decrease ambiguity and to avoid changing the meaning of the "default" `temperature` metric) I am tempted to go ahead with a more explicit `climate_actual_temperature_celsius`. But if you are absolutely sure that most Home Assistant users will understand that `climate_temperature_celsius` means current temperature, I am happy to use it instead.

Reply:
I think there is a difference between the clarity needed when looking for a Prometheus metric and when you are using Home Assistant as a GUI interface, so I think going with the `actual` is something Prometheus users would appreciate to reduce ambiguity.

Reply:
If you feel it's important to use a qualifier, I think "current" is preferable to "actual", as this matches the state attribute on the climate entity, and also seems perfectly clear from the Prometheus user's perspective. But this is just my opinion as a user of both systems.

Reply:
Good point, thanks. Given that the current metric also has `actual` in the name, I am going with `climate_current_temperature_celsius`. Sorry, @mweinelt, I think it's a better choice all things considered.